### PR TITLE
Adds Boost dependencies and basic HTTP server

### PR DESCRIPTION
**Pull request by tmccrmck**

### CODE REVIEW

File: onnxruntime/hosting/main.cc [hunk 1]
```diff
@@ -1,15 +1,73 @@
...
+
+  res.set(http::field::server, BOOST_BEAST_VERSION_STRING);
+  res.set(http::field::content_type, "plain/text");
+  res.keep_alive(context.request.keep_alive());
+  context.response = res;
+}
+
 int main(int argc, char* argv[]) {
-  using namespace onnxruntime;
+  std::string model_path;
```
Comment (non-static) [inline onnxruntime/hosting/main.cc:31]: I think it is ok to keep local variables, but consider create a configuration class to manage configurations. A TODO is ok to me here.

File: onnxruntime/hosting/main.cc [hunk 2]
```diff
@@ -1,15 +1,73 @@
...
+
+    if (vm.count("help")) {
+      std::cout << "ONNX Hosting: host an ONNX model for inferencing with ONNXRuntime\n" << std::endl
+                << desc << std::endl;
+      return EXIT_SUCCESS;
+    }
+
+    po::notify(vm); // throws on error, so do after help
+  }
+  catch(po::error& e)
```
Comment (non-static) [inline onnxruntime/hosting/main.cc]: Better having one more catch all logic, print out everything possible, then re-thrown and let the program crash.

File: onnxruntime/hosting/include/beast_http.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "util.h"
+#include "http_context.h"
+#include "routes.h"
+#include "http_session.h"
+#include "listener.h"
+
```
Comment (non-static) [inline onnxruntime/hosting/include/beast_http.h:16]: We should have namespaces

File: onnxruntime/hosting/include/beast_http.h [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+namespace net = boost::asio;       // from <boost/asio.hpp>
+using tcp = boost::asio::ip::tcp;  // from <boost/asio/ip/tcp.hpp>
+using handler_fn = std::function<void(std::string, std::string, std::string, Http_Context&)>;
+
+// Accepts incoming connections and launches the sessions
+class App {
+ public:
+  App() {
+    address_ = boost::asio::ip::make_address_v4("0.0.0.0");
+    port_ = 8080;
```
Comment (non-static) [inline onnxruntime/hosting/include/beast_http.h:31]: default value should come from a central place. But TODO for this quick PR to enable boost::beast is acceptable to me.

File: onnxruntime/hosting/include/beast_http.h [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    routes->register_controller(http::verb::post, route, fn);
+    return *this;
+  }
+
+  App& run() {
+    net::io_context ioc{threads_};
+    // Create and launch a listening port
+    std::make_shared<listener>(routes, ioc, tcp::endpoint{address_, port_})->run();
+
+    std::cout << "Listening at: \n"
```
Comment (non-static) [inline onnxruntime/hosting/include/beast_http.h:58]: Need to be updated to use logger. TODO is fine here for this PR.

File: onnxruntime/hosting/main.cc (hunk 1 above)
Comment (tmccrmck) [reply onnxruntime/hosting/main.cc:31]: I've added a TODO

File: onnxruntime/hosting/main.cc (hunk 2 above)
Comment (tmccrmck) [reply onnxruntime/hosting/main.cc]: Added one or general `std::exception` but not one with `...`

File: onnxruntime/hosting/include/beast_http.h (hunk 3 above)
Comment (tmccrmck) [reply onnxruntime/hosting/include/beast_http.h:16]: Added namespaces

File: onnxruntime/hosting/include/beast_http.h (hunk 4 above)
Comment (tmccrmck) [reply onnxruntime/hosting/include/beast_http.h:31]: Added a todo

File: onnxruntime/hosting/include/beast_http.h (hunk 5 above)
Comment (tmccrmck) [reply onnxruntime/hosting/include/beast_http.h:58]: Added a TODO

File: onnxruntime/hosting/main.cc [hunk 6]
```diff
@@ -1,16 +1,76 @@
...
 int main(int argc, char* argv[]) {
-  using namespace onnxruntime;
+  // TODO: create configuration class for all config related params
+  std::string model_path;
+  std::string address;
+  int port;
+  int threads;
 
-  SessionOptions options {};
-  InferenceSession session(options);
```
Comment (NonStatic2014) [inline onnxruntime/hosting/main.cc:10]: I would like to have both beast and onnxruntime dependencies here. We will eventually use product code but keep both here to validate build works.

**Review by NonStatic2014 — APPROVED** [review]

File: onnxruntime/hosting/main.cc (hunk 6 above)
Comment (tmccrmck) [reply onnxruntime/hosting/main.cc:10]: Good point. Added.

**Pull request merged**